src/player/Player.php

```diff
@@ -2646,6 +2659,17 @@ public function setCurrentWindow(Inventory $inventory) : bool{
 		$inventoryManager->onCurrentWindowChange($inventory);
 		$inventory->onOpen($this);
 		$this->currentWindow = $inventory;
+
+		if ($inventory instanceof SimpleInventory) {
+			if ($inventory->getMaxDistanceFromContainer() > 0) {
```

Comment:
I think we can introduce a new interface to give the `getMaxDistanceFromContainer` to avoid checking SimpleInventory which is not guaranteed